src/WebJobs.Script/FileProvisioning/PowerShell/PowerShellFileProvisioner.cs

```diff
@@ -30,22 +44,125 @@ public Task ProvisionFiles(string scriptRootPath)
 
         private void AddRequirementsFile(string scriptRootPath)
         {
-            string requirementsFilePath = Path.Combine(scriptRootPath, "requirements.psd1");
+            _logger.LogInformation($"Creating {RequirementsPsd1FileName}.");
+            string requirementsFilePath = Path.Combine(scriptRootPath, RequirementsPsd1FileName);
+
             if (!File.Exists(requirementsFilePath))
             {
-                string content = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
-                File.WriteAllText(requirementsFilePath, content);
+                string requirementsContent = FileUtility.ReadResourceString($"Microsoft.Azure.WebJobs.Script.FileProvisioning.PowerShell.requirements.psd1");
+                string guidance = null;
+
+                try
+                {
+                    string majorVersion = GetLatestAzModuleMajorVersion();
+
+                    requirementsContent = requirementsContent.Replace("# 'Az'", "'Az'");
+                    requirementsContent = requirementsContent.Replace("MAJOR_VERSION", majorVersion);
```

Comment:
Please use Regex instead of sting replace

Reply:
Fixed.